src/Simplify.elm

```diff
@@ -2876,6 +2905,126 @@ stringReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
                 checkInfo
 
 
+stringSliceChecks : CheckInfo -> List (Error {})
+stringSliceChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg, checkInfo.thirdArg ) of
+        ( Node _ (Expression.Integer 0), _, _ ) ->
+            [ Rule.errorWithFix
+                { message = "Use String.left instead"
+                , details = [ "Using String.slice with start index 0 is the same as using String.left." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy
+                    { start = checkInfo.fnRange.start
+                    , end = (Node.range checkInfo.firstArg).end
+                    }
+                    "String.left"
+                ]
+            ]
+
+        ( _, Just (Node _ (Expression.Integer 0)), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.slice with end index 0 will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( _, _, Just (Node _ (Expression.Literal "")) ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.slice on an empty string will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( start, Just end, _ ) ->
+            if Normalize.areAllTheSame checkInfo start [ end ] then
```

Comment:
This one I'm not fully sure about. We want to check if both arguments are equivalent expressions... Is "normalizing" what we are looking for here?

Reply:
Yes, that sounds good :+1: